[chores] Added firmware upgrader websocket routes

Adityashandilya555 opened the pull request:
Adds websocket routes for the OpenWISP firmware upgrader module when USE_OPENWISP_FIRMWARE environment variable is enabled.

This enables real-time upgrade progress tracking for:
- Individual upgrade operations
- Batch upgrade operations
- Device-specific upgrade progress

## Checklist

- [x] I have read the [OpenWISP Contributing Guidelines](http://openwisp.io/docs/developer/contributing.html).
- [x] I have manually tested the changes proposed in this pull request.
- [x] I have written new test cases for new code and/or updated existing tests for changes to existing code.
- [ ] I have updated the documentation.

## Reference to Existing Issue

Closes #484.

## Description of Changes
 Modified `images/common/openwisp/asgi.py` to include firmware upgrader websocket routes
Please describe these changes.

## Screenshot
<img width="635" height="121" alt="Screenshot 2025-10-26 at 12 23 40 PM" src="https://github.com/user-attachments/assets/c8b82f0e-5cfc-4b6c-a54d-dd1e5c8a6f8f" />

Please include any relevant screenshots.

Adityashandilya555 committed:
[feature] Add firmware upgrader websocket routes #484

Adds websocket routes for the OpenWISP firmware upgrader module
when USE_OPENWISP_FIRMWARE environment variable is enabled.

This enables real-time upgrade progress tracking for:
- Individual upgrade operations
- Batch upgrade operations
- Device-specific upgrade progress

Adityashandilya555 committed:
Merge branch 'master' into feature/add-firmware-upgrader-websocket-routes

Adityashandilya555 committed:
Merge branch 'master' into feature/add-firmware-upgrader-websocket-routes

nemesifier reviewed (commented):
Thanks @Adityashandilya555, but https://github.com/openwisp/openwisp-firmware-upgrader/pull/320 is not merged in master yet, so this will have to wait.

Adityashandilya555 committed:
Merge branch 'master' into feature/add-firmware-upgrader-websocket-routes

Adityashandilya555 committed:
Merge branch 'master' into feature/add-firmware-upgrader-websocket-routes

Adityashandilya555 committed:
[deps] Pin sphinx<8.0.0 for docstrfmt compatibility

Adityashandilya555 committed:
Merge branch 'master' into feature/add-firmware-upgrader-websocket-routes

Adityashandilya555 committed:
[fix] Load monitoring charts from dashboard domain

Removed OPENWISP_MONITORING_API_BASEURL = API_BASEURL so that
monitoring charts use relative URLs and load from the dashboard
domain instead of the API domain.

This fixes SSL certificate issues for users without trusted certs
on the API domain.

Closes #526

Adityashandilya555 committed:
Merge branch 'master' into feature/add-firmware-upgrader-websocket-routes

nemesifier committed:
Merge branch 'master' into feature/add-firmware-upgrader-websocket-routes

nemesifier committed:
Revert "[fix] Load monitoring charts from dashboard domain"

This reverts commit 0c13154e960304282320f70bb947be219b328102.

nemesifier committed:
[qa] Reformat
